src/monmove.cpp

```diff
@@ -1876,6 +1876,9 @@ void monster::shove_vehicle( const tripoint &remote_destination,
                 veh.skidding = true;
                 veh.velocity = shove_velocity;
                 if( shove_destination != tripoint_zero ) {
+                    if( shove_destination.z != 0 ) {
+                        veh.vertical_velocity = shove_destination.z < 0 ? -shove_velocity : +shove_velocity;
+                    }
                     g->m.move_vehicle( veh, shove_destination, veh.face );
                 }
                 veh.move = tileray( destination_delta_x, destination_delta_y );
```

Comment:
Should this use the new vehicle pointer returned by `move_vehicle`?

Reply:
it doesn't really matter. `move_vehicle` either returns null or the input vehicle. It never creates a new one. Those movement functions need a rework anyway (returning bool instead of pointers, `displace_vehicle` taking a vehicle reference instead of a point and so on).